webapp/tests/karma/js/enketo/widgets/dynamic-url.spec.ts

```diff
@@ -40,14 +40,17 @@ describe('Enketo: Dynamic URL Widget', () => {
     expect($(DynamicUrlWidget.selector).attr('href')).to.equal('http://google.com?q=');
   });
 
-  it('should update the href for the dynamically generated URL when the dynamic value changes', () => {
+  it('should update the href for the dynamically generated URL when the dynamic value changes', (done) => {
     buildHtml('helloWorld');
 
     new DynamicUrlWidget($(DynamicUrlWidget.selector)[0]);
     const dynamic = 'worldHello';
     $('.url span').text(dynamic);
 
-    expect($(DynamicUrlWidget.selector).attr('href')).to.equal(`http://google.com?q=${dynamic}`);
+    setTimeout(() => {
+      expect($(DynamicUrlWidget.selector).attr('href')).to.equal(`http://google.com?q=${dynamic}`);
+      done();
+    }, 0);
```

Comment:
So just for the record (for anyone that is looking into this in the future) the normal Angular way of testing async UI updates via `fakeAsync` does not seem to work here.  And from reading online this seems to be expected (I guess the MutationObserver falls outside of what Angular has control of).  In general the consensus seems to be against trying too hard to unit test MutationObserver behavior.  However, since this works (at least for now) I say we ship it as is!  :+1: